homeassistant/components/syncthing/sensor.py

```diff
@@ -266,3 +283,66 @@ def _filter_state(self, state):
         state["label"] = self._folder_label
 
         return state
+
+
+class SpeedSensor(CoordinatorEntity):
+    """A syncthing speed sensor."""
+
+    NAMES = {"download": "inBytesTotal", "upload": "outBytesTotal"}
+
+    def __init__(self, coordinator, url, server_id, name, version):
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._name = name
+        self._last_total_value = self._current_value()
+        self._last_updated = monotonic()
+        self._state = 0
+
+        short_server_id = server_id.split("-")[0]
+        self._attr_name = f"{short_server_id} {self._name}"
+        self._attr_unique_id = f"{short_server_id}-{self._name}"
+        self._attr_icon = "mdi:speedometer"
+        self._attr_unit_of_measurement = DATA_RATE_MEGABYTES_PER_SECOND
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, server_id)},
+            "name": f"Syncthing ({url})",
+            "manufacturer": "Syncthing Team",
+            "sw_version": version,
+            "entry_type": "service",
+        }
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if not self.available:
+            return
+        return self._state
+
+    @property
+    def device_info(self):
+        """Return device information."""
+        return
+
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        return {"total": self._current_value()}
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        current = self._current_value()
+        now = monotonic()
+        speed = (
```

Comment:
I don't think we should add this sensor at all, as it is not a value provided by the device or service. Instead, this is a virtual value calculated right here in Home Assistant. For those cases, we have other integration available if users want that information.